### ISSUE TITLE
Set badge on Samsung devices running Android 7

### ISSUE THREAD

**Issue by phobeous** [Pushwoosh/pushwoosh-appcelerator-titanium#48]
I'm using titanium 3.3.2 module for titanium platform (latest version as of writing). The exception that is shown in logcat when trying to set badge is:

[ERROR] Pushwoosh: [ShortcutBadger] Unable to execute badge [ERROR] Pushwoosh: com.pushwoosh.thirdparty.a.b: Unable to execute badge

So I've realized that PushWoosh is using ShortcutBadger library to set badges in android devices. I've installed ShortcutBadger tester (https://play.google.com/store/apps/details?id=me.leolin.isbadgeworking) and it works like a charm. I've tested on Galaxy J3 (android 5.1.1), S6 (android 7) and S8 (android 7), getting successful results. But my app, which relies on PushWoosh module is not working S6 and S8 (both with android 7). As far as I can guess, PW module is using an older version of ShorcutBadger library.

**Comment by wfhm** [comment]
Hi,

This is a known issue, which was fixed in Pushwoosh native Android SDK already, however the Titanium module was not yet updated. We will release a new version shortly.

**Comment by phobeous** [comment]
Great! I hope that new version to be published ASAP. I guess Android SDK issue is https://github.com/Pushwoosh/pushwoosh-android-sdk/issues/55
It would be fantastic if titanium module was updated with each native Android/iOS/WindowsPhone update (at least Android & iOS)

**Comment by safadig** [comment]
@phobeous That can be done if we have a Hyperloop version of module...Since Hyperloop is now part of Free/Indie may be its worth while..It is way beyond my capabilities but hope someone can

**Comment by phobeous** [comment]
@safadig Yes, but using latest PW android SDK would probably fall into changing our current module usage, even with HyperLoop, as it has breaking changes (from 4.12.1 to 5.0.4). I tried using latest .jar from PW (5.0.4) but classes and methods are different. PW module uses source version of ShortcutBadger library. If it would used a .jar inside /lib folder in the module, we could have updated ourselves (probably). I've developed several android modules for titanium and I'm just expecting for the chance to use Hyperloop (I've got an Indie lic).

Let's see if PW updates titanium module with backward compatibility or we'll probably have to update our titanium client code.
